In [ ]:
# Import libraries
from pyspark.sql import SparkSession
from pyspark.sql.functions import col, from_json, when, size, explode, expr
from pyspark.sql.types import ArrayType, StringType, StructType, StructField, IntegerType, DoubleType, MapType
import time

student_id = "CH24M000"  # your roll number, letters must be in capital letter
app_name = student_id + "_Assignment_1"
assignment_no = "Assignment_1"

spark = SparkSession.builder \
    .appName(app_name) \
    .config("spark.executor.instances", "1") \
    .config("spark.executor.memory", "2G") \
    .config("spark.executor.cores", "2") \
    .config("spark.driver.memory", "3G") \
    .getOrCreate()
# File paths (unique for each student)
# base path to access the dataset from server. Uncomment when submitting the job to server to use the spark cluster
base_path = f"/opt/spark/data/{student_id}/"

# base path to access the sample dataset from your google drive
# base_path = "PATH"


# Load CSV files
customers_df = spark.read.option("multiline", True) \
                         .option("escape", "\"") \
                         .option("quote", "\"") \
                         .option("header", True) \
                         .csv(base_path + "customers.csv")
products_df = spark.read.option("multiline", True) \
                         .option("escape", "\"") \
                         .option("quote", "\"") \
                         .option("header", True) \
                         .csv(base_path + "products.csv")
orders_df = spark.read.option("multiline", True) \
                         .option("escape", "\"") \
                         .option("quote", "\"") \
                         .option("header", True) \
                         .csv(base_path + "orders.csv")



In [ ]:
from pyspark.sql import SparkSession

spark = SparkSession.builder.appName("CH24M000").getOrCreate()
input_path = "/opt/spark/data/CH24M000/Assignment_1/data.csv"
df = spark.read.option("header", "true").csv(input_path)
df.show()


In [5]:
import requests
import pprint

# Read code from the file
with open("CH24M000.py", "r") as file:
    code_content = file.read()

url = "https://lab.samsai.io/submit"

# DO NOT MODIFY
headers = {
    "Authorization": "Bearer f9318293d4e405e5cff5d03a348a02ae0c4331916cd390041a700045d1bcb16a",
    "Content-Type": "application/json"
}
payload = {
    "student_id": "CH24M000",
    "code": code_content
}

response = requests.post(url, headers=headers, json=payload)

print("Status Code:", response.status_code)
print("Response Body:", response.text)



Status Code: 200
Response Body: {"command":"docker exec -i spark-master /opt/spark/bin/spark-submit --master spark://spark-master:7077 --conf spark.executor.memory=4g --conf spark.driver.memory=2g --conf spark.sql.adaptive.enabled=true --conf spark.sql.adaptive.coalescePartitions.enabled=true /opt/spark/code/7834b89d-a091-456c-ad47-354fa80e3c0f.py","dag_url":"https://lab.samsai.io/spark-ui","history_url":"https://lab.samsai.io/history","job_id":"7834b89d-a091-456c-ad47-354fa80e3c0f","log_url":"https://lab.samsai.io/logs/7834b89d-a091-456c-ad47-354fa80e3c0f.log","script_path":"/opt/spark/code/7834b89d-a091-456c-ad47-354fa80e3c0f.py","status":"submitted","student_id":"CH24M000"}



Status Code: 200
Response Body: {"command":"docker exec -i spark-master /opt/spark/bin/spark-submit --master spark://spark-master:7077 --conf spark.executor.memory=4g --conf spark.driver.memory=2g --conf spark.sql.adaptive.enabled=true --conf spark.sql.adaptive.coalescePartitions.enabled=true /opt/spark/code/96d07c81-c0a8-4f99-b5cc-83f8b84fed2f.py","dag_url":"https://lab.samsai.io/spark-ui","history_url":"https://lab.samsai.io/history","job_id":"96d07c81-c0a8-4f99-b5cc-83f8b84fed2f","log_url":"https://lab.samsai.io/logs/96d07c81-c0a8-4f99-b5cc-83f8b84fed2f.log","script_path":"/opt/spark/code/96d07c81-c0a8-4f99-b5cc-83f8b84fed2f.py","status":"submitted","student_id":"CH24M000"}

In [4]:
!curl https://lab.samsai.io/logs/96d07c81-c0a8-4f99-b5cc-83f8b84fed2f.log

Command: docker exec -i spark-master /opt/spark/bin/spark-submit --master spark://spark-master:7077 --conf spark.executor.memory=4g --conf spark.driver.memory=2g --conf spark.sql.adaptive.enabled=true --conf spark.sql.adaptive.coalescePartitions.enabled=true /opt/spark/code/96d07c81-c0a8-4f99-b5cc-83f8b84fed2f.py
25/06/04 11:42:31 INFO SparkContext: Running Spark version 3.4.0
25/06/04 11:42:31 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
25/06/04 11:42:31 INFO ResourceUtils: ==============================================================
25/06/04 11:42:31 INFO ResourceUtils: No custom resources configured for spark.driver.
25/06/04 11:42:31 INFO ResourceUtils: ==============================================================
25/06/04 11:42:31 INFO SparkContext: Submitted application: CH24M000
25/06/04 11:42:31 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory